### PR TITLE
adding caster slot spells table

### PR DESCRIPTION
**Pull request by gmriggs**
This PR builds on https://github.com/ACEmulator/ACE/pull/3206

Just some preliminary data definitions, no logic actually adding the 'first slot' spells to casters yet

### CODE REVIEW

**Comment by gmriggs** [comment]
Requesting review from @ziang4891 and @harliq

File: Source/ACE.Server/Factories/Enum/WealthRating.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        Undef,
+        Shoddy,
+        Poor,
+        Medium,
+        Good,
+        Rich,
+        Incomparable,
+        Exceptional,
+        Phenomenal
+    }
```
Comment (harliq) [inline Source/ACE.Server/Factories/Enum/WealthRating.cs:14]: Maybe Comment what tiers are for each of those for others?

Comment (gmriggs) [reply Source/ACE.Server/Factories/Enum/WealthRating.cs:14]: I think WealthRating might be used for different things

For this initial function that was added, yes it directly maps tiers to wealth ratings. ie. 8 wealth ratings, 8 tiers

**Review by ziang4891 — APPROVED** [review]

**Review by LtRipley36706 — APPROVED** [review]

**Review by ziang4891 — APPROVED** [review]

**Review by harliq — DISMISSED** [review]

**Review by LtRipley36706 — DISMISSED** [review]

File: Source/ACE.Server/Factories/LootGenerationFactory.cs [hunk 2]
```diff
@@ -1391,37 +1391,142 @@ private static int GetValue(int tier, int work, double gemMod, double matMod)
...
+
+            // only reduce value?
+            if (iValue < wo.Value)
+                wo.Value = iValue;
+        }
+
+        private static void MutateValue_Spells(WorldObject wo)
+        {
+            if (wo.ItemMaxMana != null)
+                wo.Value += wo.ItemMaxMana;
```
[the remark this answers is not in the document]
Comment (gmriggs) [reply Source/ACE.Server/Factories/LootGenerationFactory.cs]: yes, it should actually be ItemMaxMana * 2 apparently

**Review by ziang4891 — APPROVED** [review]

**Review by LtRipley36706 — APPROVED** [review]

**Pull request merged**